src/Components/Footer/Footer.tsx

```diff
@@ -0,0 +1,107 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+import { deleteTodo } from '../../api/todos';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  filter: Filter;
+  setFilter: (filter: Filter) => void;
+  setTodos: (todos: Todo[]) => void;
+  filteredTodos: Todo[];
+  todos: Todo[];
+  setErrors: (errors: Errors) => void;
+  setWaitingResponse: (id: number[]) => void;
+};
+
+export const Footer: FC<Props> = ({
+  filter,
+  setFilter,
+  setTodos,
+  filteredTodos,
+  todos,
+  setErrors,
+  setWaitingResponse,
+}) => {
+  const clearCompleted = async () => {
+    let completed = filteredTodos.filter((todo) => todo.completed);
+
+    completed = filteredTodos.filter((todo) => todo.completed);
+
+    const completedIds = completed.map((todo) => todo.id!);
```

Comment:
Don't do like that
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91411692/04142a3a-d328-4697-97ed-6de543c2aabf)
```suggestion
    const completedIds = completed.map((todo) => todo.id);
```